src/QuorumTools/Control.hs

```diff
@@ -51,25 +64,115 @@ event e = do
   let fire = void $ tryPutMVar mvar e
   pure (occurred, fire)
 
--- | Creates a stream of values for a many publishers to update, and many
+-- | Creates a stream of values for many publishers to update, and many
 -- consumers to subscribe (to value changes) or observe the current value.
-behavior :: (MonadIO m, Monoid a) => m (Behavior a)
+behavior :: (MonadIO m) => m (Behavior a)
 behavior = liftIO $ atomically $ Behavior <$> newTChan <*> newTMVar mempty
 
-transition :: MonadIO m => Behavior a -> (a -> a) -> m ()
+transition :: MonadIO m => Behavior a -> (Last a -> a) -> m ()
 transition (Behavior tc tm) f = liftIO $ atomically $ do
-  prev <- takeTMVar tm
-  let next = f prev
-  putTMVar tm next
-  writeTChan tc next
+  v <- takeTMVar tm
+  let v' = f v
+  putTMVar tm $ Last $ Just v'
+  writeTChan tc v'
+
+subscribe' :: Behavior a -> STM (TChan a)
```

Comment:
The idea was that the prime variants were more composable and less useful to end users, but good point. Renaming these.